worker/rhpv2.go

```diff
@@ -913,6 +907,8 @@ func RPCSettings(ctx context.Context, t *rhpv2.Transport) (settings rhpv2.HostSe
 		return rhpv2.HostSettings{}, err
 	} else if err := json.Unmarshal(resp.Settings, &settings); err != nil {
 		return rhpv2.HostSettings{}, fmt.Errorf("couldn't unmarshal json: %w", err)
+	} else if err := validateNetAddress(ctx, settings.NetAddress); err != nil {
```

Comment:
I'm not sure this is the right place for the validation considering that the outcome of this will determine whether a host is online or offline. I feel like this is more of a gouging thing that people might later tweak to allow for private addresses for example.

Reply:
Hm damn, I really like where the check is at. It gets rid of a couple of manual `NetAddress` checks in various places and instead errors out with a super meaningful and helpful error message.